### ISSUE TITLE
Object2IntHashMap incorrectly increments it size on each merge call

### ISSUE THREAD

**Issue by zyulyaev** [real-logic/agrona#303]
For example:
```
var map = new Object2IntHashMap<String>(0);

map.merge("foo", 1, Integer::sum);
map.merge("foo", 1, Integer::sum);

assert map.size() == 1; // fails since the size == 2
map.values().forEach(System.out::println); // throws since it can't find second element
```

See https://github.com/real-logic/agrona/blob/f6c6d8be2cef3318d99ce31c347a3844d3c2acaa/agrona/src/main/java/org/agrona/collections/Object2IntHashMap.java#L483

**Comment by vyazelenko** [comment]
The fix is also available in [1.23.1](https://github.com/real-logic/agrona/releases/tag/1.23.1) patch release.

**Comment by zyulyaev** [comment]
Cool, thank you!
